docs/source/hub.rst

```diff
@@ -4,6 +4,227 @@
+|topic          |String          |The topic URL that the subscriber wishes to subscribe to or unsubscribe from. Topic |
+|               |                |is an URL to any resource that notifies the Hub about updates.                      |
++---------------+----------------+------------------------------------------------------------------------------------+
+|leaseSeconds   |String          |Number of seconds for which the subscriber would like to have the subscription      |
+|               |                |active. After this time pass subscription need to be renewed.                       |
++---------------+----------------+------------------------------------------------------------------------------------+
+|secret         |String          |A subscriber-provided secret string that will be used to compute an HMAC digest for |
+|               |                |authorized content distribution.                                                    |
++---------------+----------------+------------------------------------------------------------------------------------+
+|hub.challenge  |                |The parameter sent by the Hub in GET request in process of intent verification.     |
+|               |                |The subscriber must respond with HTTP success (2xx) status with challenge parameter |
+|               |                |in response body to confirm (un)subscription.                                       |
++---------------+----------------+------------------------------------------------------------------------------------+
+
+##########
+Publishing
+##########
+
+Whenever there is an update in topic, the publisher notifies the hub providing the resource URL (topic) which
+is updated. Then hub fetches the updated content and distributes it to all the active subscribers of that topic.
+
+REST CALL
+---------
+
+To notify about update by REST call send HTTP ``POST`` request to
+
+    ::
+
+        http://{motech-server}/{motech-war-file}/module/hub/?&hub.mode={hubMode}&hub.url={url}
+
+with content type set to ``application/x-www-form-urlencoded``, ``{hubMode}`` to literal string "publish" and
+``{url}`` to updated topic URL.
+
+OSGI Service
+------------
+
+The Hub module expose OSGi ``ContentDistributionService`` interface for publishing:
+
+.. code-block:: java
+
+    public interface ContentDistributionService {
+
+        void distribute(String url);
+
+    }
+
+To publish with this service update your topic, then just call distribute method providing updated topic URL.
+
+##############
+Error handling
+##############
+
+Errors that occurs when using Hub module are wrapped in a custom ``HubException`` exception class. This class contains
+``reason`` field that contain description of what caused failure and ``HubErrors`` object which contains more detailed
+information about error, which at least are ``message``, ``code`` and ``httpStatus``. Please refer to table shown below
```

Comment:
I don't get this part.. Do you mean that the field contains information about the message, code and http status for which the error occured?